src/libraries/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -516,6 +529,11 @@ internal void WriteCentralDirectoryFileHeader()
                 extraFieldLength = (ushort)bigExtraFieldLength;
             }
 
+            if (_hasUnicodeEntryNameOrComment)
+                _generalPurposeBitFlag |= BitFlagValues.UnicodeFileNameAndComment;
+            else
+                _generalPurposeBitFlag &= ~BitFlagValues.UnicodeFileNameAndComment;
```

Comment:
OK, I guess this is the reason why you stopped setting this flag in set_FullName.

Reply:
Yep :)